lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp

```diff
@@ -1520,6 +1534,35 @@ void HdVP2Mesh::_UpdateDrawItem(
     });
 }
 
+void HdVP2Mesh::_HideAllDrawItems(const TfToken& reprToken) {
+    HdReprSharedPtr const& curRepr = _GetRepr(reprToken);
+    if (!curRepr) {
+        return;
+    }
+
+    _MeshReprConfig::DescArray reprDescs = _GetReprDesc(reprToken);
+
+    // For each relevant draw item, update dirty buffer sources.
+    int drawItemIndex = 0;
+    for (size_t descIdx = 0; descIdx < reprDescs.size(); ++descIdx) {
+        const HdMeshReprDesc& desc = reprDescs[descIdx];
+        if (desc.geomStyle == HdMeshGeomStyleInvalid) {
+            continue;
+        }
+
+        auto* drawItem = static_cast<HdVP2DrawItem*>(curRepr->GetDrawItem(drawItemIndex++));
+        if (!drawItem)
+            continue;
+        MHWRender::MRenderItem* renderItem = drawItem->GetRenderItem();
+        if (!renderItem)
+            continue;
+
+        _delegate->GetVP2ResourceRegistry().EnqueueCommit([renderItem]() {
+            renderItem->enable(false);
```

Comment:
Setting enable can change which world the renderItem in, we need those changes to be serialized so use the commit system to do it from the main thread.

Reply:
Soon we will need ThreadRun to free main-thread and get processing of commits started as soon as possible.